process/geometry/utils.py

```diff
@@ -0,0 +1,120 @@
+"""
+Module to hold plotting functions, used in plot_proc.py, which are common to multiple reactor components
+"""
+import numpy as np
+
+
+def plotdh(r0: float, a: float, triang: float, kap: float) -> tuple:
+    """Plots half a thin D-section, centred on z = 0
+
+    :param r0: major radius of centre
+    :type r0: float
+    :param a: horizontal radius
+    :type a: float
+    :param triang: plasma triangularity
+    :type triang: float
+    :param kap: plasma elongation
+    :type kap: float
+    :return: radial and vertical coordinates of D-section
+    :rtype: tuple
+    """
+    angs = np.linspace(0, np.pi, 50, endpoint=True)
+    rs = r0 + a * np.cos(angs + triang * np.sin(1.0 * angs))
+    zs = kap * a * np.sin(angs)
+    return rs, zs
+
+
+def plotdhgap(
+    inpt: float,
+    outpt: float,
+    inthk: float,
+    outthk: float,
+    toppt: float,
+    topthk: float,
+    triang: float,
+) -> tuple:
+    """Plots a half thick D-section with a gap
+
+    :param inpt: inner point
+    :type inpt: float
+    :param outpt: outer point
+    :type outpt: float
+    :param inthk: inner thickness
+    :type inthk: float
+    :param outthk: outer thickness
+    :type outthk: float
+    :param toppt: top point
+    :type toppt: float
+    :param topthk: top thickness
+    :type topthk: float
+    :param triang: plasma triangularity
+    :type triang: float
+    :return: radial and vertical coordinates of D-section
+    :rtype: tuple
+    """
+    arc = np.pi / 4.0
+    r01 = (inpt + outpt) / 2.0
+    r02 = (inpt + inthk + outpt - outthk) / 2.0
+    a1 = r01 - inpt
+    a2 = r02 - inpt - inthk
+    kap1 = toppt / a1
+    kap2 = (toppt - topthk) / a2
+    angs = np.linspace(0.0, (np.pi / 2.0) - arc / 2.0, 50, endpoint=True)
+    rs1 = r01 + a1 * np.cos(angs + triang * np.sin(angs))
+    zs1 = kap1 * a1 * np.sin(angs)
+    rs2 = r02 + a2 * np.cos(angs + triang * np.sin(angs))
+    zs2 = kap2 * a2 * np.sin(angs)
+    angs = np.linspace(np.pi, np.pi + ((np.pi / 2.0) - arc), 50, endpoint=True)
+    rs3 = r01 + a1 * np.cos(angs + triang * np.sin(angs))
+    zs3 = kap1 * a1 * np.sin(angs)
+    rs4 = r02 + a2 * np.cos(angs + triang * np.sin(angs))
+    zs4 = kap2 * a2 * np.sin(angs)
+
+    return rs1, rs2, rs3, rs4, zs1, zs2, zs3, zs4
+
+
+def ellips_fill(
+    a1: float = 0,
+    a2: float = 0,
+    b1: float = 0,
+    b2: float = 0,
+    x0: float = 0,
+    y0: float = 0,
+    ang1: float = 0,
+    ang2: float = np.pi / 2,
+) -> list:
+    """Fills the space between two concentric ellipse sectors
```

Comment:
I think it more returns the vertices of a shape which, when filled... "fills the space between two concentric ellipse sectors"